htdocs/contact/card.php

```diff
@@ -1295,14 +1296,36 @@
 		$linkback = '<a href="'.DOL_URL_ROOT.'/contact/list.php?restore_lastsearch_values=1">'.$langs->trans("BackToList").'</a>';
 
 		$morehtmlref = '<div class="refidno">';
-		if (empty($conf->global->SOCIETE_DISABLE_CONTACTS))
-		{
+
+		// Code added here, resolving issue when users who have no access rights to see for fournisseurs can access to them anyway by passing trough contacts  //
+		$sql = "SELECT fournisseur, client FROM " . MAIN_DB_PREFIX . "societe WHERE rowid = " . (int) $object->socid ;
```

Comment:
Addin a requets here is useless. When you make $objsoc->fetch($object->socid); later, you will get the property ->client and ->fournisseur. So you can add a test after. 
Also note that not having permission to see supplier means we are not allowed to see information about the supplier (for example prices and references) but the permission to know if user can see the third-party is user->rights->societe->lire.
Also changing the link will not fix trouble if user enter url manually. So i pushed a better fix taking all of this into consideration.